src/FrameManager.ts

```diff
@@ -360,6 +360,80 @@ export class FrameManager extends EventEmitter {
   }
 }
 
+/**
+ * At every point of time, page exposes its current frame tree via the [page.mainFrame()](#pagemainframe) and [frame.childFrames()](#framechildframes) methods.
+ *
+ * [Frame] object's lifecycle is controlled by three events, dispatched on the page object:
+ * - ['frameattached'](#event-frameattached) - fired when the frame gets attached to the page. A Frame can be attached to the page only once.
+ * - ['framenavigated'](#event-framenavigated) - fired when the frame commits navigation to a different URL.
+ * - ['framedetached'](#event-framedetached) - fired when the frame gets detached from the page.  A Frame can be detached from the page only once.
```

Comment:
Once these are documented on the `Page` class we can link to them but we won't be able to yet - so maybe just make these:

. `framedetached` - fires when the ...